### PR TITLE
ci: github action for nightly docker

### PR DESCRIPTION
**Pull request by juliushaertl**
Change-Id: I56d28f4b2428b248c52d550825d1ac337a536bdf

### Summary

Provides a separate docker build method to build a container isolated from the host system. The goal is to have a nightly docker image on the github container registry that can be used for easier testing and running automated tests in integrators.

For now only the built from the precompiled core assets is covered.

Note I've tested building in the github action in my local fork https://github.com/juliushaertl/online/actions/runs/10083478724 but not in here yet.

### Checklist

- [x] I have run `make prettier-write` and formatted the code.
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Build nightly docker image
+
+on:
+  schedule:
+    - cron: "30 0 * * *"
+  push:
```
Comment (juliushaertl) [inline .github/workflows/docker.yml:6]: To be removed before merge as then schedule will trigger only on the default github branch

**Comment by juliushaertl** [comment]
@pedropintosilva Maybe you can request review if anyone else is also relevant for this

**Comment by caolanm** [comment]
I wonder if we should have an --enable-experimental in here somewhere by default in the online configure line, or use ONLINE_EXTRA_BUILD_OPTIONS of --enable-experimental in the build.sh because I think we enable that for our normal code builds

**Comment by juliushaertl** [comment]
Sure if it is closer to what code builds. I can add that.

**Pull request merged**